src/modules/xkeyboard.cpp

```diff
@@ -284,42 +284,45 @@ namespace modules {
 
       size_t size = vec.size();
       if (size != 2 && size != 3) {
-        m_log.warn("%s: malformed layout-icon '%s'", name(), it);
+        m_log.warn("%s: Malformed layout-icon '%s'", name(), it);
         continue;
       }
 
       const string& layout = vec[0];
       if (layout.empty()) {
-        m_log.warn("%s: layout-icon '%s' is invalid: there must always be a layout in icon declaration", name(), it);
+        m_log.warn("%s: layout-icon '%s' is invalid: there must always be a layout defined", name(), it);
         continue;
       }
 
-      if (size == 2) {
-        if (layout == drawtypes::layouticonset::VARIANT_ANY) {
-          throw module_error(
-              "Using '" + it + "' for layout-icon means declaring a default icon, use 'layout-icon-default' instead");
-        }
+      string variant;
+      string icon;
 
-        define_layout_icon(it, layout, drawtypes::layouticonset::VARIANT_ANY, std::make_shared<label>(vec[1]));
+      if (size == 2) {
+        variant = layouticonset::VARIANT_ANY;
+        icon = vec[1];
       } else {
-        const string& variant = vec[1];
-        if (layout == drawtypes::layouticonset::VARIANT_ANY && variant == drawtypes::layouticonset::VARIANT_ANY) {
-          throw module_error(
-              "Using '" + it + "' for layout-icon means declaring a default icon, use 'layout-icon-default' instead");
-        }
+        variant = vec[1];
+        icon = vec[2];
+
         if (variant.empty()) {
-          define_layout_icon(it, layout, drawtypes::layouticonset::VARIANT_NONE, std::make_shared<label>(vec[2]));
-        } else {
-          define_layout_icon(it, layout, variant, std::make_shared<label>(vec[2]));
+          variant = layouticonset::VARIANT_NONE;
         }
       }
+
+      if (layout == layouticonset::VARIANT_ANY && variant == layouticonset::VARIANT_ANY) {
+        m_log.warn("%s: Using '%s' for layout-icon means declaring a default icon, use 'layout-icon-default' instead",
+            name(), it);
+      }
```

Comment:
Having removed the throw here means that the user will:
* either get 2 warning messages if a default  icon is already defined: 1 here and 1 from define_layout_icon
* have another way to define a default icon if none already defined and you seem to not want that

Reply:
You're right. I forgot to `continue` in that case

Reply:
`continue` restored